auraed/src/runtime/cells/cells.rs

```diff
@@ -45,35 +48,40 @@ pub(crate) struct CellsTable {
 // - Get Cgroup from pid
 // - Get Cgroup and pids from executable_name
 
-impl CellsTable {
-    /// Add the [cell] to the cache with key [cell_name].
-    /// Returns an error if a duplicate [cell_name] already exists in the cache.
-    pub async fn insert<T: Into<Cell>>(
+impl Cells {
+    /// Add the [Cell] to the cache with key [CellName].
+    /// Returns an error if a duplicate [CellName] already exists in the cache.
+    pub async fn allocate<T: Into<Cell>>(
         &self,
         cell_name: CellName,
         cell: T,
     ) -> Result<()> {
         let mut cache = self.cache.lock().await;
-
-        // TODO: replace with this when it becomes stable
-        // cache.try_insert(cell_name.clone(), cgroup)
-
-        // Check if there was already a cgroup in the table with this cell name as a key.
-        if cache.contains_key(&cell_name) {
-            return Err(CellsError::CellExists { cell_name });
-        }
-        // Ignoring return value as we've already assured ourselves that the key does not exist.
-        let _ = cache.insert(cell_name, cell.into());
+        let _ = allocate(&mut cache, cell_name, cell)?;
```

Comment:
on the offchance that a future change copies and mutates "cell" in `allocate` then we might want to return the Cell here from the function call instead of ignoring it.

Reply:
We can't return `Cell` or a ref to it here because the lifetime of it is tied to the lock (line above). That was also why we needed `allocate_then` which takes a `Fn` that lets the caller have access to `&Cell`.